bittensor/_receptor/receptor_impl.py

```diff
@@ -267,7 +267,7 @@ def prerequisite_check(self, request):
             return False, request
         
         # ---- Check endpoint----
-        if self.endpoint.uid == -1:
+        if self.endpoint.hotkey  == 'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX':
```

Comment:
This is a bit weird too, looks like it's checking if we are in test mode. We shouldn't be "checking" if we are in test mode, should just act upon it instead.

Reply:
Its not so much of a testing endpoint but a 'null' endpoint. We put the check here because it keeps all the other code the same, rather than say filtering at a higher level, you actually query the null endpoint as usual.